File: frontend/types/inventory-overview.ts
import * as z from "zod";

export const InventoryCostingOverviewSchema = z.object({
  monthYear: z.string(),
  incomingCount: z.number(),
  incomingTotalCost: z.number(),
  outgoingCount: z.number(),
  outgoingTotalCost: z.number(),
});

export type InventoryCostingOverviewValues = z.infer<
  typeof InventoryCostingOverviewSchema
>;

File: backend/types/inventory-overview.go
package types

// InventoryCostingOverview is a struct to map database table inventory overview.
type InventoryCostingOverview struct {
	MonthYear         string  `json:"monthYear" db:"month_year"`
	IncomingCount     int     `json:"incomingCount" db:"incoming_count"`
	IncomingTotalCost float64 `json:"incomingTotalCost" db:"incoming_total_cost"`
	OutgoingCount     int     `json:"outgoingCount" db:"outgoing_count"`
	OutgoingTotalCost float64 `json:"outgoingTotalCost" db:"outgoing_total_cost"`
}
